doc/tor.md

```diff
@@ -40,7 +45,11 @@ outgoing connections, but more is possible.
 	-onlynet=onion  Make outgoing connections only to .onion addresses. Incoming
 	                connections are not affected by this option. This option can be
 	                specified multiple times to allow multiple network types, e.g.
-	                ipv4, ipv6, or onion.
+	                ipv4, ipv6 or onion. If you use this option with values other
+	                than onion you *cannot* disable onion connections; outgoing onion
+	                connections will be enabled when you use -proxy or -onion. Use
+	                -noonion or -onion=0 if you want to be sure there are no outbound
+	                onion connections over the default proxy or your defined -proxy.
```

Comment:
Does this mean `onlynet=ipv4` `onion=0` does not disable onion connections? This seems confusing.

`-noonion` or `onion=0` is always needed to explicitly disable outbound onion access, as described in the `-proxy` and `onion` section?

Reply:
(I missed this last commit, glad you asked for the clarification.  I ACKd 42e61805734c36da7a11586aabc7fd5742dba267)

Reply:
> Does this mean onlynet=ipv4 onion=0 does not disable onion connections? This seems confusing.

Yup, a default fallback outbound pure .onion not IP name resolving proxy just for onion address will be created by default even without -proxy definition  if onion!=0 and regardless of -onlynet= if -listenonion=1 ( default) if the inbound Tor onion service could be created successfully by the ever  polling torcontroller in bitcoind   (that is your  advertised Tor addr created by bitcoind independent of what onion services defined in torrc ) The idea was, (i guess)  if that is successfully to create also an outbound proxy to reach Tor onion nodes.

So suppose u have a running bitcoind daemon and edit at some point just torrc to allow the control port or start the Tor service, bitcoind does if -listeonion=1 permanent poll for such changes and instantly creates on the fly an inbound onion address and an outbound proxy.  So if u want to be sure that the outbound default proxy is disabled since u want i.e. just ipv4 set -noonion since -onlynet=ipv4 wont do that and please note the inbound Tor is created regardless of that if -listenonion=1 what is the default.  

If that is a desired way to act depends, fixes or changes to that are not merged yet and at least that's the way it is and its now documented so that users are aware of that.

Reply:
@jonasnick  did edit the comment text to be more precise, so please review ur thumbs up ;-)

Reply:
Dropping this commit.

Reply:
@jonatack  i am almost certain there will be an rc5 and some backport's ping @luke-jr, so no time to rush here and have another release that is not precise in what the options and Tor-controller does.

Do u need a detailed descriiption and  log to show the real behavior?

Reply:
@Rspigler tyi, only explicit -nonion or -onion=0 can prevent outbound onion connections, solely -onllynet=pv4/ipv6 is not sufficient to disable that.

that what is said in the PR now dropped.

```
Please use -noonion or -onion=0 if you want to be sure to have no
outbound onion connections over the default proxy or your defined proxy..
```

https://github.com/bitcoin/bitcoin/commit/654dc04b67a44199352a1b25494c9d617cd090bc#diff-a33b1b15efbc14c65460ef7a883b530ca52144a4f06642e4a188351d959f9894R42

Reply:
@Saibato I might propose the dropped commits, including yours, in a new PR and make some of the changes suggested by @vasild.

Reply:
@Saibato thanks for the explanation/edits.  Although now dropped, would ACK in a new PR

Reply:
> @Saibato thanks for the explanation/edits. Although now dropped, would ACK in a new PR

Done in #20757, please add suggestions or ACKs there.